typing/typedtree.mli

```diff
@@ -76,10 +75,10 @@ and 'k pattern_desc =
   (* value patterns *)
   | Tpat_any : value pattern_desc
         (** _ *)
-  | Tpat_var : Ident.t * string loc -> value pattern_desc
+  | Tpat_var : Ident.t * string loc * Types.value_mode -> value pattern_desc
         (** x *)
   | Tpat_alias :
-      value general_pattern * Ident.t * string loc -> value pattern_desc
+      value general_pattern * Ident.t * string loc * Types.value_mode -> value pattern_desc
```

Comment:
Quick question: In general I kind of "understand" why we keep the mode for patterns variables and aliases - however, it seems that they are not technically needed? I don't see them being used anywhere.

Reply:
These are needed, they're extracted by `let_bound_idents_with_modes` (some of whose callers want the value modes)

Reply:
I see - thanks!